webapp/src/workspace.ts

```diff
@@ -43,8 +43,12 @@ function lookup(id: string) {
     return allScripts.filter(x => x.header.id == id || x.header.path == id)[0]
 }
 
-export function gitsha(data: string) {
-    return (sha1("blob " + U.toUTF8(data).length + "\u0000" + data) + "")
+export function gitsha(data: string, encoding: "utf-8" | "base64" = "utf-8") {
+    if (encoding == "base64") {
+        const d = atob(data);
+        return (sha1("blob " + d.length + "\u0000" + d) + "")
```

Comment:
still no luck with this SHA @mmoskal

Reply:
It seems crypto-js is doing UTF8 parse. I think we may need something like this:

```
sha1(CryptoJS.enc.Utf8.parse("blob " + len + "\u0000").concat(
CryptoJS.enc.Base64.parse(data)))
```​

https://cryptojs.gitbook.io/docs/#encoders